src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,82 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import translate from 'amo/i18n/translate';
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import type { ResultShelfType } from 'amo/reducers/home';
+
+import './styles.scss';
+
+export const MOZILLA_USER_ID = config.get('mozillaUserId');
+
+type Props = {|
+  i18n: Object,
```

Comment:
Because `i18n` isn't a prop that is passed directly to this component from a parent (but rather included via `translate()`), it should go into a type called `InternalProps`. See other components for an example of this. It also has its own type.